src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -217,46 +227,52 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_aggregate_node(
 
     for (auto& aggregate_expression : aggregate_expressions) {
-      Assert(aggregate_expression->expression_list().size(), "Aggregate: empty expression list");
+      Assert(aggregate_expression->aggregate_function_arguments().size(), "Aggregate: empty expression list");
       DebugAssert(aggregate_expression->type() == ExpressionType::Function, "Expression is not a function.");
 
       // Do not project for COUNT(*)
       if (aggregate_expression->aggregate_function() == AggregateFunction::Count &&
-          (aggregate_expression->expression_list())[0]->type() == ExpressionType::Star) {
+          (aggregate_expression->aggregate_function_arguments())[0]->type() == ExpressionType::Star) {
         continue;
       }
 
       // Add original expression of the function to the Projection.
-      column_expressions.emplace_back(aggregate_expression->expression_list()[0]);
+      Assert(aggregate_expression->aggregate_function_arguments().size() == 1,
+             "Aggregate functions with more than one argument not supported right now");
+      const auto argument_lqp_expression =
+          std::dynamic_pointer_cast<OperatorExpression>(aggregate_expression->aggregate_function_arguments()[0]);
+      DebugAssert(argument_lqp_expression, "Expression in AggregateNode was not an LQPExpression");
 
-      // Create a ColumnReference expression for the column id of the Projection.
-      const auto column_ref_expr = Expression::create_column(ColumnID{current_column_id});
-      current_column_id++;
+      projection_expressions.emplace_back(argument_lqp_expression);
 
       // Change the expression list of the expression representing the aggregate.
-      aggregate_expression->set_expression_list({column_ref_expr});
+      aggregate_expression->set_aggregate_function_arguments(
+          {OperatorExpression::create_column(ColumnID{current_column_id})});
+      current_column_id++;
     }
 
-    aggregate_input_operator = std::make_shared<Projection>(aggregate_input_operator, column_expressions);
+    aggregate_input_operator = std::make_shared<Projection>(aggregate_input_operator, projection_expressions);
   }
 
   /**
    * 2. Build Aggregate
    */
-  std::vector<AggregateDefinition> aggregate_definitions;
+  std::vector<AggregateColumnDefinition> aggregate_definitions;
   aggregate_definitions.reserve(aggregate_expressions.size());
 
   for (const auto& aggregate_expression : aggregate_expressions) {
     DebugAssert(aggregate_expression->type() == ExpressionType::Function, "Only functions are supported in Aggregates");
 
     const auto aggregate_function_type = aggregate_expression->aggregate_function();
-    const auto root_expr = (aggregate_expression->expression_list())[0];
+    const auto argument_expression =
+        std::dynamic_pointer_cast<OperatorExpression>(aggregate_expression->aggregate_function_arguments()[0]);
+    DebugAssert(argument_expression, "Couldn't cast Expression to OperatorExpression.");
```

Comment:
This error message is too close to the implementation and redundant. Something like "The aggregate function argument is not an expression"?

Reply:
Na, that wouldn't have been the case here. Sometimes you have these asserts that you can just explain with "WTF, this is a unexpected and some sort of bug." - that's why I'll change it to "Wrong expression type found in LQP. Bug."